pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -257,3 +303,79 @@ func (cd *clusterDeploy) getYAML(cluster *v3.Cluster, agentImage, authImage stri
 
 	return buf.Bytes(), err
 }
+
+func (cd *clusterDeploy) getClusterAgentImage(name string) (string, error) {
+	uc, err := cd.clusterManager.UserContext(name)
+	if err != nil {
+		return "", err
+	}
+
+	d, err := uc.Apps.Deployments("cattle-system").Get("cattle-cluster-agent", v1.GetOptions{})
+	if err != nil {
```

Comment:
is this logic also used to deploy the deployment/daemonset for the first time? if so, we need to handle the NotFound error, and return `""` ?

Reply:
we should only ever hit this logic if we have just deployed a cluster or are recaching the agents after a restart. with our logic it shouldnt get a notfound but definite need to account for it just in case and will add, good catch.